test/AutoRest.TestServer.Tests/Mgmt/TestProjects/ExactMatchInheritanceTests.cs

```diff
@@ -28,6 +28,7 @@ public ExactMatchInheritanceTests()
         [TestCase(typeof(object), typeof(ExactMatchModel9))]
         [TestCase(typeof(object), typeof(ExactMatchModel7))]
         [TestCase(typeof(object), typeof(ExactMatchModel11))]
+        [TestCase(typeof(object), typeof(AzureResourceFlattenModelData))]
```

Comment:
need to add a flattenModel that should inherit from one of the reference types as well such as TrackedResource<T>